src/xrpld/app/tx/detail/SetTrust.cpp

```diff
@@ -243,6 +243,26 @@ SetTrust::doApply()
     bool const bSetFreeze = (uTxFlags & tfSetFreeze);
     bool const bClearFreeze = (uTxFlags & tfClearFreeze);
 
+    bool granularDelegated = false;
+    if (ctx_.isDelegated && !ctx_.gpSet.empty())
+    {
+        granularDelegated = true;
+        // If granular permission is delegated under the TrustSet transaction.
+        // Currently we only support TrustlineAuthorize, TrustlineFreeze and
+        // TrustlineUnfreeze granular permission.
+        if (bSetNoRipple || bClearNoRipple || bQualityIn || bQualityOut)
+            return terNO_AUTH;
+        if (bSetAuth &&
+            ctx_.gpSet.find(gpTrustlineAuthorize) == ctx_.gpSet.end())
```

Comment:
Can simplify here and below:

    if (bSetAuth &&
            !ctx_.gpSet.contains(gpTrustlineAuthorize))